src/test/java/integration/TestRequestManager.java

```diff
@@ -0,0 +1,425 @@
+        }
+        sesSto.removePropertyChangeListener(listener);
+        Assert.assertTrue(subscribedIds.containsAll(projectIds));
+    }
+
+    private void testUnsubscribeFromProject(long projId) throws InterruptedException {
+        logger.info("Unsubscribe from project");
+        Semaphore waiter = new Semaphore(0);
+        PropertyChangeListener listener = (event) -> {
+            if (event.getPropertyName().equals(SessionStorage.SUBSCRIBED_PROJECTS)) {
+                if (event.getOldValue().equals(projId)) {
+                    waiter.release();
+                }
+            }
+        };
+        sesSto.addPropertyChangeListener(listener);
+        reqMgr.unsubscribeFromProject(projId);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+        sesSto.removePropertyChangeListener(listener);
+    }
+
+    private void testFetchProjects() throws InterruptedException {
+        logger.info("Fetching projects");
+        Semaphore waiter = new Semaphore(0);
+        PropertyChangeListener listener = (event) -> {
+            if (event.getPropertyName().equals(SessionStorage.PROJECT_LIST)) {
+                List<Project> projects = (List<Project>) event.getNewValue();
+                List<Long> ids = projects.stream().map(Project::getProjectID).collect(Collectors.toList());
+                List<Long> correctProjects = new ArrayList<>();
+                correctProjects.add(proj1.getProjectID());
+                correctProjects.add(proj2.getProjectID());
+                if (ids.containsAll(correctProjects)) {
+                    waiter.release();
+                }
+            }
+        };
+        sesSto.addPropertyChangeListener(listener);
+        reqMgr.fetchProjects();
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+        sesSto.removePropertyChangeListener(listener);
+    }
+
+    private void testSubscribeToProject(long projId) throws InterruptedException {
+        logger.info("Subscribing to project");
+        Semaphore waiter = new Semaphore(0);
+        PropertyChangeListener listener = (event) -> {
+            if (event.getPropertyName().equals(SessionStorage.SUBSCRIBED_PROJECTS)) {
+                long id = (long) event.getNewValue();
+                if (id == projId) {
+                    waiter.release();
+                }
+            }
+        };
+        sesSto.addPropertyChangeListener(listener);
+        reqMgr.subscribeToProject(projId);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+        sesSto.removePropertyChangeListener(listener);
+    }
+
+    private void testDeleteProject(long projId) throws InterruptedException {
+        logger.info("Deleting project");
+        Semaphore waiter = new Semaphore(0);
+        PropertyChangeListener listener = (event) -> {
```

Comment:
Intellij is flagging this block as duplicate code, with testRemoveSelfFromProject. Are they testing exactly the same thing? Are there any differences?